certification/internal/shell/base_on_ubi_test.go

```diff
@@ -3,42 +3,70 @@ package shell
 import (
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
+	"github.com/redhat-openshift-ecosystem/openshift-preflight/cli"
+	log "github.com/sirupsen/logrus"
 )
 
+// podmanEngine is a package-level variable. In some tests, we
+// override it with a "happy path" engine, that returns good data.
+// In the unhappy path, we override it with an engine that returns
+// nothing but errors.
+
 var _ = Describe("BaseOnUBI", func() {
 	var (
-		engine      FakePodmanEngine
-		base_on_ubi BaseOnUBICheck
+		baseOnUbiCheck BaseOnUBICheck
+		originalEngine cli.PodmanEngine
+		fakeEngine     cli.PodmanEngine
 	)
 
 	BeforeEach(func() {
-		engine = FakePodmanEngine{
+		fakeEngine = FakePodmanEngine{
 			RunReportStdout: `ID="rhel"
 NAME="Red Hat Enterprise Linux"
 `,
 			RunReportStderr: "",
 		}
+		originalEngine = podmanEngine
+	})
+	AfterEach(func() {
+		podmanEngine = originalEngine
 	})
```

Comment:
Topic for thought: This block basically becomes a required part of every series of tests in this suite, right? So every check that leverages the FakePodmanEngine will need to do this BeforeEach and AfterEach task to make sure they're not inadvertently touching the package-level variable and impacting other tests?

Again - nothing to change here. Just trying to zoom out a bit. Feel free to resolve whenever.

Reply:
I will address that in a different PR. (Probably #2) once this lands.